whois-api/src/main/java/net/ripe/db/whois/api/rest/WhoisRestService.java

```diff
@@ -301,6 +313,8 @@ public Response lookup(
             queryBuilder.addFlag(QueryFlag.NO_FILTERING);
         }
 
+        addBasic(request, passwords);
+
```

Comment:
Do we allow both basic authentication and password as query as a parameter in the same request or it should be either one of them ? What happens when basic authentication is failed but password in query parameter is correct or vice-versa?

Reply:
We should give priority to basic authentication and ignore passwords as query parameter as we plan to send warning to user to switch to basic auth. If basic auth passwords are wrong it should failed authentication

Reply:
I think that this is a question for Ed. I think that if the user is passing one correct password we shouldn't return a 401 even if the basicAuth is not correct...We are limiting the use of the passwords query parameter which should still have the same behaviour as before (we are just deprecating it, we shouldn't modify anything there)

Reply:
If we are  saying password as a query parameter is deprecated and user should use basic auth then we should always give priority to basicvAuth and ignore password as parameter if basicAuth is available else it is confusing and error prone and makes it less secure

Reply:
lets discuss with Ed, I see your point of view but Ed told me to add the basicAuth to passwords...not to overlap passwords with basic. This is easy to do with the current code, so let's see what he says because I think that each of you have a different idea.

Reply:
Don't ignore `password` yet we can't deprecate it until we've informed users and see decreased usage, the impact is too great otherwise

Reply:
Just consider basic as another `password` for now